CORE-5182 Add helm chart for app simulator

williamvigorr3 opened the pull request:
Add helm charts so the app simulator can be deployed into a Corda worker cluster.

-  There are separate deployment templates for the `receiver`, `sender` (when sending messages in `CONTINUOUS` mode) and `db_sink`. As well as a Job template for the `sender` (when sending messages in `ONE_OFF` mode). I added parameters for these to applications/tools/p2p-test/app-simulator/charts/app-simulator/values.yaml, with some sensible defaults. There are also parameters in the values.yaml file so that the app simulator can connect to the app simulator database and kafka. All these parameters can be overridden on the command line or with another yaml file.
- I have added a Job template to create `p2p.app.received_msg` topic. This is almost the same as the job under the main corda chart. I had to change the topic creation tool (tools/plugins/topic-config/src/main/kotlin/net/corda/cli/plugins/topicconfig/Create.kt) to be able to create the test schema, to do this I changed the package under which the test schema is stored here: https://github.com/corda/corda-api/pull/544.
- I changed to the app simulator so that all options can be supplied on the command line, as well as via a config file. 
This makes it easy to pass in command line options via helm into the dock container.

Merge https://github.com/corda/corda-api/pull/544 first.

To use the app simulator
====

Deploy 3 worker clusters using
```
export U=wvigor
chart_version=0.1.0-alpha.1662143290096
docker_image_version=5.0.0.0-alpha-1662141681234
declare -a namespaces=("$U-cluster-a" "$U-cluster-mgm" "$U-cluster-b")

for namespace in ${namespaces[@]}; do
  echo Creating $namespace
  kubectl create ns $namespace

  echo Installing prereqs into $namespace
  helm upgrade --install prereqs -n $namespace \
    oci://corda-os-docker.software.r3.com/helm-charts/corda-dev \
    --set image.registry="corda-os-docker.software.r3.com" \
    --set kafka.replicaCount=3,kafka.zookeeper.replicaCount=1 \
    --render-subchart-notes \
    --timeout 10m \
    --wait

 echo Installing corda into $namespace
 helm upgrade --install corda -n $namespace oci://corda-os-docker.software.r3.com/helm-charts/corda --set "imagePullSecrets={docker-registry-cred}"  --set image.tag=$docker_image_version --set image.registry="corda-os-docker.software.r3.com" --values values.yaml --values debug.yaml --wait --version $chart_version
done

kubectl port-forward --namespace $U-cluster-a deployment/corda-rpc-worker 8888 &
kubectl port-forward --namespace $U-cluster-b deployment/corda-rpc-worker 8889:8888 &
kubectl port-forward --namespace $U-cluster-mgm deployment/corda-rpc-worker 8890:8888 &
```

Run the MultiClusterDynamicNetworkTest. This is disabled by default and some variables inside: `applications/workers/release/rpc-worker/build.gradle` need to be set:
```
export E2E_CLUSTER_A_RPC_HOST=localhost
export E2E_CLUSTER_A_RPC_PORT=8888
export E2E_CLUSTER_B_RPC_HOST=localhost
export E2E_CLUSTER_B_RPC_PORT=8889
export E2E_CLUSTER_C_RPC_HOST=localhost
export E2E_CLUSTER_C_RPC_PORT=8890
export E2E_CLUSTER_A_P2P_HOST="corda-p2p-gateway-worker.wvigor-cluster-a"
export E2E_CLUSTER_A_P2P_PORT=8080
...
```
To run the test:
```
./gradlew :applications:workers:release:rpc-worker:e2eTest --tests "net.corda.applications.workers.rpc.MultiClusterDynamicNetworkTest"
```

Deploy the app simulator postgres database into it's own namespace:
```
cd applications/tools/p2p-test/app-simulator/charts/app-simulator-db
kubectl create ns $U-db
helm upgrade --install -n $U-db db ./ --render-subchart-notes
```
Extract the password:
```
export POSTGRES_ADMIN_PASSWORD=$(kubectl get secret --namespace $U-db db-postgresql -o jsonpath="{.data.postgres-password}" | base64 --decode)
```
(Note that the app simulator database helm chart doesn't give the simulator user enough permissions to be able to write to the sent_messages_table.

Deploy a receiving and db_sink app simulator into the $U-cluster-b namespace:
```
cd applications/tools/p2p-test/app-simulator/charts/app-simulator
helm upgrade --install app-simulator ./ -n $U-cluster-b --set db.appSimulator.namespace=$U-db --set db.appSimulator.user=postgres --set db.appSimulator.password=$POSTGRES_ADMIN_PASSWORD --set appSimulators.receiver.enabled=true --set appSimulators.dbSink.enabled=true  --set "imagePullSecrets={docker-registry-cred}" --set image.tag=5.0.0.0-alpha-1663077102624
```
`kubectl get pods  --all-namespaces` should show something like:
```
NAMESPACE     NAME                                             READY   STATUS             RESTARTS         AGE
cluster-b         app-simulator-create-topics-lzjqx                0/1     Completed          0                145m
cluster-b         app-simulator-db-sink-6f4fbf44f8-rrbbv           1/1     Running            0                145m
cluster-b         app-simulator-receiver-676fc96bb9-sck9k          1/1     Running            0                145m
cluster-b         corda-create-topics-lr682                        0/1     Completed          0                26h
cluster-b         corda-crypto-worker-84f7cd8666-r9lqh             1/1     Running            0                26h
cluster-b         corda-db-worker-5659c898bb-h9wwp                 1/1     Running            0                26h
cluster-b         corda-flow-worker-5d7c56f77-lrksm                1/1     Running            0                26h
cluster-b         corda-membership-worker-5bc48bd74-qvm4f          1/1     Running            0                26h
cluster-b         corda-p2p-gateway-worker-787487bf58-vjnzf        1/1     Running            0                26h
cluster-b         corda-p2p-link-manager-worker-5c89b948c8-cjrj7   1/1     Running            253 (21s ago)    26h
cluster-b         corda-rpc-worker-6c78ccbb86-bmlr4                1/1     Running            0                26h
cluster-b         corda-setup-db-7jx8p                             0/1     Completed          0                26h
cluster-b         kafka-deployment-6bcf67786-s2w48                 1/1     Running            0                28h
cluster-b         prereqs-postgresql-0                             1/1     Running            0                28h
cluster-b         zookeeper-deployment-5886f98b8d-ktxwh            1/1     Running            0                28h
db            db-postgresql-0                                  1/1     Running            5 (3h38m ago)    3h40m
...
```

First can extract the x500Names and groupId's from the cluster by looking in the Link Manager pod logs for both `$U-cluster-a`  (our) and `$U-cluster-b` (peer). For lines which look like:
```
11:08:28.482 [durable processing thread inbound_message_processor_group-link.in] INFO  net.corda.p2p.linkmanager.sessions.SessionManagerImpl {} - Outbound session 2b05b145-707d-46ce-9378-d38ad1d18cbb established (local=HoldingIdentity(x500Name=OU=E2eClusterImpl-1-1662116846286, O=Bob, L=London, C=GB, groupId=2e916688-ddd4-40ee-b4df-6f25a4496c66), remote=HoldingIdentity(x500Name=OU=E2eClusterImpl-1-1662116846287, O=Mgm, L=London, C=GB, groupId=2e916688-ddd4-40ee-b4df-6f25a4496c66)).
```
Currently you need to remove the spaces from the x500Name and also replace `,` with `\,`.

Deploy a sending app simulator into $U-cluster-a namespace:
```
helm upgrade --install app-simulator ./ -n $U-cluster-a --set db.appSimulator.user=postgres  --set db.appSimulator.namespace=$U-db --set db.appSimulator.user=postgres --set db.appSimulator.password=$POSTGRES_ADMIN_PASSWORD --set appSimulators.sender.enabled=true --set "imagePullSecrets={docker-registry-cred}" --set image.tag=5.0.0.0-alpha-1663077102624 --set "appSimulators.sender.ourX500Name=OU=E2eClusterImpl-1-1663090181629\,O=Alice\,L=London\,C=GB" --set appSimulators.sender.peerX500Name="OU=E2eClusterImpl-1-1663090181725\,O=Bob\,L=London\,C=GB" --set "appSimulators.sender.ourGroupId=a525a03e-ab4a-427c-b35b-c0d6642c98ab" --set "appSimulators.sender.peerGroupId=a525a03e-ab4a-427c-b35b-c0d6642c98ab"
```

This will send 50 messages from `OU=E2eClusterImpl-1-1662116846172,O=Alice,L=London,C=GB` to `OU=E2eClusterImpl-1-1662116846286,O=Bob,L=London,C=GB`.

williamvigorr3 committed:
CORE-5182 Add AppSimulator helm chart

williamvigorr3 committed:
Fix command line options for App Simulator

williamvigorr3 committed:
Add helm charts for db-sink and sender

williamvigorr3 committed:
Use alpha corda api version

williamvigorr3 committed:
Update moved file

williamvigorr3 committed:
Fix detekt and update readme

williamvigorr3 committed:
Only json deserialize messages from the app simulator and mount kafka certs correctly

williamvigorr3 committed:
Merge remote-tracking branch 'origin/release/os/5.0' into WillV/CORE-5182-Add-Helm-Chart-for-App-Simulator-release

williamvigorr3 committed:
Bump version

williamvigorr3 committed:
Split on the messageId not the key

williamvigorr3 committed:
Remove log line

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/NOTES.txt:
```diff
@@ -0,0 +1 @@
+1. Add Notes
```
Not sure who that's an instruction to!

williamvigorr3 replied:
Me :). I added some notes.

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/_helpers.tpl:
```diff
@@ -0,0 +1,212 @@
+{{- end }}
+
+{{/*
+Volume mounts for corda workers
```
nit: not Corda workers

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/_helpers.tpl:
```diff
@@ -0,0 +1,212 @@
+{{- end }}
+
+{{/*
+Volumes for corda workers
```
nit: not Corda workers

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/app-simulator-dbsink.yaml:
```diff
@@ -0,0 +1,62 @@
+{{- if .Values.appSimulators.dbSink.enabled }}
+{{- $_ := set . "type" "dbSink" }}
```
I'm not particularly proud of having introduced this global variable hack in the Corda Helm chart and, given the limited usage of the type in this chart, if you were to just inline it in the places that it's used, that would make me happy.

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/topic-job.yaml:
```diff
@@ -0,0 +1,99 @@
+{{- if .Values.bootstrap.kafka.enabled }}
```
Why is the topic bootstrapping duplicated here?

williamvigorr3 replied:
This is removed now.

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/values.yaml:
```diff
@@ -0,0 +1,195 @@
+      # -- The time after (in ISO-8601 duration format) which the message will expire in seconds, if message expiry is enabled. This is used
+      # to calculate the TTL of the generated messages, which corresponds to the current time plus the duration specified by `expireAfterTime`.
+      expireAfterTime: "PT180S"
+    debug:
```
You expose a debug port but do you ever actually enable debug on the JVM?

williamvigorr3 replied:
I added this.

davidcurrie commented on tools/plugins/topic-config/src/main/kotlin/net/corda/cli/plugins/topicconfig/Create.kt:
```diff
@@ -33,6 +33,12 @@ class Create(
     var partitionOverride: Int = 1
 
+    @CommandLine.Option(
+        names = ["-s", "--schema-name"],
```
Not overly keen on exposing this given this is meaningless to a real user. If I understand correctly, does this mean that we're ignoring all of the normal topics and just creating the one test topic instead? Is there a way we could just refactor the topic creation so that we could re-use the required part of it programmatically from the app simulator rather than through the API? Or even creating a separate CLI plugin?

williamvigorr3 replied:
I will try move the topic creation into the app-simulator. That is probably easier than, adding a cli plugin. Also we won't need another docker image.

davidcurrie replied:
Thanks, @williamvigorr3. I hope I'm not being too bloody minded about something that seems so trivial but I know that exposing this sort of test flag on a product can be a bit of a slippery slope.

williamvigorr3 replied:
Doing that will mean I can remove the topic bootstrap, so I think it is a good idea.

williamvigorr3 replied:
I managed to move it.

yift-r3 commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/Chart.yaml:
```diff
@@ -0,0 +1,24 @@
+name: appSimulator
+description: A Helm chart for the application simulator
+
+# A chart can be either an 'application' or a 'library' chart.
```
Do we need all those comments?

williamvigorr3 replied:
These are generated by default by helm. They explain what the yaml is doing. I think we should leave them.

yift-r3 commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/NOTES.txt:
```diff
@@ -0,0 +1 @@
+1. Add Notes
```
?

williamvigorr3 replied:
I have now added some notes.

yift-r3 commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/_helpers.tpl:
```diff
@@ -0,0 +1,212 @@
+{{/*
```
We have so many of those, and they all look the same, is there a way to include one tpl from another?

davidcurrie replied:
Do you mean from one chart to another? That's the purpose of a library chart.

yift-r3 replied:
Yes, that would be nice.

williamvigorr3 replied:
I agree the amount of code duplication between `applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/_helpers.tpl` and `charts/corda/templates/_helpers.tpl` is not ideal. I don't like the idea of downloading a corda library chart here (with some old version) as the corda chart is also stored in this repo. 

What about instead using `template` to refer to the main corda _helpers.tpl in this _helpers.tpl? I don't know if that will/can work.

davidcurrie replied:
I don't think it will work. I think we just accept the duplication here.

yift-r3 replied:
:(

yift-r3 commented on applications/tools/p2p-test/app-simulator/README.md:
````diff
@@ -150,6 +150,9 @@ These -m options are passed into the Kafka client. For example to use TLS to con
 java -jar ./applications/p2p-link-manager/build/bin/corda-p2p-link-manager*.jar -msecurity.protocol=SSL -mssl.truststore.location=/certs/ca.crt -mssl.truststore.type=PEM
 ```
+It is possible to pass in the all the options from the command line, instead of using a config file.
+The dbParams can be specified with `-d` options e.g. `-dusername=root -dpassword=securePassword` and loadGenerationParams with `-l` options e.g. `-lpeerX500Name=Me`.
````
There are a few places that we validate the X500 names - The README should probably have a valid name.

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -99,43 +95,45 @@ class AppSimulator @Activate constructor(
-        val simulatorMode = simulatorConfig.getEnum(SimulationMode::class.java, "simulatorMode")
+        val configFromFile = parameters.simulatorConfig?.let { ConfigFactory.parseFile(it) } ?: ConfigFactory.empty()
+        val clients = parameters.clients ?: configFromFile.getIntOrNull(PARALLEL_CLIENTS_KEY) ?: DEFAULT_PARALLEL_CLIENTS
+        val simulatorMode = parameters.simulationMode ?: configFromFile.getEnumOrNull("simulatorMode")
```
Can't the `configFromFile`, `clients` and `simulatorMode` be lazy members of the `CliParameters`

williamvigorr3 replied:
I think doing that makes understanding the app simulator more difficult. Although it would probably reduce the amount and complexity of code. At the moment the program has a distinct phase where we parse and check all the arguments. We do this before we start any subscriptions or spawn any threads. 

If we check the an arguments after we start a subscription and it is invalid then the messaging library might retry several times (depends how quickly `shutDownService.shutdown(FrameworkUtil.getBundle(this::class.java))` happens).

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -99,43 +95,45 @@ class AppSimulator @Activate constructor(
     ) {
-        val connectionDetails = readDbParams(simulatorConfig)
-        val loadGenerationParams = readLoadGenParams(simulatorConfig)
+        val sendTopic = parameters.sendTopic ?: P2P_OUT_TOPIC
```
Same as before, can't this be calculated in the parameters (or another class that wrap the parameters)?

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -177,57 +176,60 @@ class AppSimulator @Activate constructor(
         resources.add(receiver)
     }
 
+    @Suppress("LongParameterList")
```
The `parameters` has the `instanceId`, we can remove that parameter (at least)

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -177,57 +176,60 @@ class AppSimulator @Activate constructor(
+        val host = getDbParameter("host", config, parameters, ::logErrorAndShutdown) ?: return null
+        val db = getDbParameter("db", config, parameters, ::logErrorAndShutdown) ?: return null
+        val dbParams = DBParams(username, password, host, db)
+        logger.info("$dbParams")
```
Do we need this?

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -177,57 +176,60 @@ class AppSimulator @Activate constructor(
-        val loadGenerationType =
-            config.getEnum(LoadGenerationType::class.java, "$LOAD_GEN_PARAMS_PREFIX.loadGenerationType")
+    private fun readLoadGenParams(configFromFile: Config, parameters: CliParameters): LoadGenerationParams {
+        val peerX500Name = getLoadGenStrParameter("peerX500Name", configFromFile, parameters, ::logErrorAndShutdown)
```
Do we want to make sure it's not null?

williamvigorr3 replied:
I improved this code. We check it is not null inside `getLoadGenStrParameter`, which throws an exception if there is a problem. This is caught and logged.

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulator.kt:
```diff
@@ -177,57 +176,60 @@ class AppSimulator @Activate constructor(
+    private fun readLoadGenParams(configFromFile: Config, parameters: CliParameters): LoadGenerationParams {
+        val peerX500Name = getLoadGenStrParameter("peerX500Name", configFromFile, parameters, ::logErrorAndShutdown)
+        val peerGroupId = getLoadGenStrParameter("peerGroupId", configFromFile, parameters, ::logErrorAndShutdown)
+        val ourX500Name = getLoadGenStrParameter("ourX500Name", configFromFile, parameters, ::logErrorAndShutdown)
```
Do we want to validate the x500 names?

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/ArgParsingUtils.kt:
```diff
@@ -0,0 +1,182 @@
+            configFromFile: Config,
+            parameters: CliParameters, logErrorAndShutdown: (String) -> Unit
+        ): E? {
+            val stringParameter = getParameter(
```
Can we use `getLoadGenStrParameter`?

williamvigorr3 replied:
I can list the possible options if I don't.

yift-r3 commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/Receiver.kt:
```diff
@@ -65,12 +65,18 @@ class Receiver(private val subscriptionFactory: SubscriptionFactory,
-                Record(destinationTopic, messageReceivedEvent.messageId, objectMapper.writeValueAsString(messageReceivedEvent))
+                val messageId = authenticatedMessage.header.messageId
+                //Only JSON deserialize messages from another app-simulator (not sent by the MGM for example).
+                if (messageId.split(":").last().toIntOrNull() != null) {
```
Can we use the `subsystem` instead of id?

williamvigorr3 replied:
Good idea

williamvigorr3 committed:
Create app received messages topic inside the AppSimulator

Instead of from a K8s topic creation job.

williamvigorr3 committed:
Merge remote-tracking branch 'origin/release/os/5.0' into WillV/CORE-5182-Add-Helm-Chart-for-App-Simulator-release

williamvigorr3 committed:
Use AppSimulator subsystem

williamvigorr3 committed:
Respond to review comments on charts

williamvigorr3 committed:
Add NOTES.txt

Also add option to set total number of messages and turn off writting
to db for sender.

williamvigorr3 committed:
Fix broken test

williamvigorr3 committed:
Enable debug in app simulator helm chart

Also fix bug in Kafka sasl authentication

williamvigorr3 committed:
Respond to review comments

williamvigorr3 committed:
Revert changes to values

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/NOTES.txt:
```diff
@@ -0,0 +1,64 @@
+{{- if and (.Values.appSimulators.sender.enabled) (eq .Values.appSimulators.sender.loadGenerationType "ONE_OFF") }}
```
Nice use of NOTES.txt 👍

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/templates/_helpers.tpl:
```diff
@@ -0,0 +1,160 @@
+{{/*
+Volumes for the appSimulator
+*/}}
+{{- define "appSimulator.workerVolumes" }}
```
```suggestion
{{- define "appSimulator.volumes" }}
```

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/values.yaml:
```diff
@@ -0,0 +1,163 @@
+  registry: "corda-os-docker.software.r3.com"
+  # -- image repository
+  repository: "corda-os-app-simulator"
+  # Overrides the image tag whose default is the chart appVersion.
```
```suggestion
  # -- overrides the image tag whose default is the chart appVersion.
```

davidcurrie commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/values.yaml:
```diff
@@ -0,0 +1,163 @@
+  # -- the default CPU/memory resource request for the app simulator containers
+  requests: {}
+
+# Logging configuration
+logging:
+  # -- log format; "json" or "text"
+  format: "json"
+  # -- log level; one of "all", "trace", "debug", "info", "warn" (the default), "error", "fatal", or "off"
+  level: "warn"
```
is this used?

davidcurrie reviewed (commented):
A few nits but looking good from the Helm chart side...

williamvigorr3 committed:
Respond to review comments

davidcurrie reviewed (dismissed):
👍

williamvigorr3 committed:
Move arg reading code into parameter data classes

dimosr commented on applications/tools/p2p-test/p2p-layer-deployment/src/main/kotlin/net/corda/p2p/deployment/commands/KafkaSetup.kt:
```diff
@@ -41,7 +40,6 @@ class KafkaSetup(
             "cleanup" to mapOf("policy" to "delete"),
         )
         private val topicsToConfig = mapOf(
-            APP_RECEIVED_MESSAGES_TOPIC to simpleConfig,
```
Could this cause issues in the `p2p-layer-deployment` tooling? If so, should we remove this as part of https://r3-cev.atlassian.net/browse/CORE-5783?

dimosr replied:
On that note, did you verify `p2p-layer-deployment` is still functional?

williamvigorr3 replied:
Yes. I tested the p2p layer deployment, with the topic creation inside the app simulator.

dimosr replied:
ah, right - the simulator will be creating the topic itself so removing it from this one shoudn't cause issues

williamvigorr3 committed:
Fix detekt and build with new version of Corda API

williamvigorr3 committed:
Merge remote-tracking branch 'origin/release/os/5.0' into WillV/CORE-5182-Add-Helm-Chart-for-App-Simulator-release

dimosr commented on applications/tools/p2p-test/app-simulator/charts/app-simulator/values.yaml:
```diff
@@ -0,0 +1,156 @@
+# app simulator configuration
+appSimulators:
+  sender:
+    enabled: true
```
Should we switch `enabled` to false as default for all the modes?

It might make it slightly easier when running as well, e.g. you will only have to enable what you want (without having to explicitly disable the rest).

williamvigorr3 replied:
Let's do that.

dimosr commented on applications/tools/p2p-test/app-simulator/src/main/kotlin/net/corda/p2p/app/simulator/AppSimulatorTopicCreator.kt:
```diff
@@ -0,0 +1,42 @@
+{
+    companion object {
+        internal const val APP_RECEIVED_MESSAGES_TOPIC = "p2p.app.received_msg"
+        private const val APP_RECEIVED_MESSAGES_PARTITIONS = 10
+        private const val APP_RECEIVED_MESSAGES_REPLICATION = 1
+        private val APP_RECEIVED_MESSAGE_TOPIC_CONF = mapOf(
```
[As discussed briefly in the standup] I think we need a way to adjust those e.g. for scenarios where we deploy 3 brokers (and we want a more realistic 3-way replication) and when doing scalability testing (where we'll want to increase the number of partitions).

But, if it doesn't affect the demo, we can leave that as a follow-up - maybe add a small ticket under https://r3-cev.atlassian.net/browse/CORE-5814?

williamvigorr3 replied:
Added https://r3-cev.atlassian.net/browse/CORE-6699

dimosr commented:
One more thing on:

> Merge https://github.com/corda/corda-api/pull/544 first.

Is this necessary? Given these changes are backwards compatible if done the other way around (i.e. remove all usages of this topic and then remove `TestSchema`), should we do them the other way around (merge this PR first and then merge the corda-api as a clean-up) to skip doing a version bump, taking locks etc.?

williamvigorr3 commented:
> One more thing on:
> 
> > Merge [corda/corda-api#544](https://github.com/corda/corda-api/pull/544) first.
> 
> Is this necessary? Given these changes are backwards compatible if done the other way around (i.e. remove all usages of this topic and then remove `TestSchema`), should we do them the other way around (merge this PR first and then merge the corda-api as a clean-up) to skip doing a version bump, taking locks etc.?

I don't know. I think technically we need to bump the version when we delete the code in the api repo (we don't have to do this now though). Otherwise we might end up in the situation where if we checkout an old commit before this change it won't build against the latest build of corda-api (with a specific version).

dimosr commented:
ah, I see what you mean. Indeed, a version bump is needed if we want to protect against the scenario you described. But, to avoid blocking this until you get the lock for a version bump, can we switch it to use version 185 and then do the version bump separately as a follow-up (tiny) PR? Conscious there's a bit of queue atm..

dimosr commented:
> Note that the app simulator database helm chart doesn't give the simulator user enough permissions to be able to write to the sent_messages_table.

Don't we need to change this as part of this change?

williamvigorr3 commented:
> > Note that the app simulator database helm chart doesn't give the simulator user enough permissions to be able to write to the sent_messages_table.
> 
> Don't we need to change this as part of this change?

At the moment I am using the admin user `postgres` instead. So I can test this change without it. Want me to raise a Jira to fix this?

williamvigorr3 committed:
Respond to review comments

williamvigorr3 committed:
Merge remote-tracking branch 'origin/release/os/5.0' into WillV/CORE-5182-Add-Helm-Chart-for-App-Simulator-release

williamvigorr3 committed:
Use release version of corda-api

dimosr commented:
> At the moment I am using the admin user postgres instead. So I can test this change without it. Want me to raise a Jira to fix this?

Given this is mostly targeted for testing, I am ok with us using the admin user for the time being.

williamvigorr3 committed:
Fix values.yaml after testing

dimosr reviewed (dismissed):
LGTM (good to get a confirmation from @yift-r3 the changes he requested around config are OK) and also I think we need to re-point to the DP2 branch and merge there now.

williamvigorr3 committed:
Merge remote-tracking branch 'origin/release/os/5.0-DevPreview2' into WillV/CORE-5182-Add-Helm-Chart-for-App-Simulator-retarget
